### PR TITLE
Delete and Edit records

### PR DESCRIPTION
**Pull request by sangeethailango**

### CODE REVIEW

File: lib/eco_service_web/live/eco_service_live/community_list.html.heex [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
-    <.link phx-click="delete" phx-value-id={community.id}> <span class="text-blue-500">Delete</span> </.link>    
-</:col>
+  <:col :let={community} label="Community Name"> <%= community.name%> </:col>
+  <:col :let={community} label="Location Area Zone"> <%= community.location_area_zone%> </:col>
+  <:col :let={community} label="Delete"> 
+      <.link phx-click="delete" phx-value-id={community.id} > <span class="text-blue-500">Delete</span> </.link>    
+  </:col>
+  <:col :let={community} label="Edit"> 
+      <.link href={~p"/community/#{community.id}/edit"} > <span class="text-blue-500">Edit</span> </.link>    
+  </:col>
```
Comment (nittin-shankar) [inline lib/eco_service_web/live/eco_service_live/community_list.html.heex:10]: Why do you want to have two columns in the table for the action buttons(edit and delete)? The general practice we've all followed is to have all the actions in a single column. 

Besides, why do you have a label for the column of action buttons? It's not at all a recommended practice to have a column titled delete and have many delete buttons in that column. The UX can quickly get very messy. Your UX is honestly looking very messy. Please fix that!

CleanShot 2024-04-12 at 1  59 59@2x

File: lib/eco_service_web/live/eco_service_live/community_list.html.heex [hunk 2]
```diff
@@ -1,14 +1,35 @@
+<h1 class="font-bold text-lg text-center">List of communities</h1>
 <.table id="community" rows={@communities}>
-<:col :let={community} label="Community Name"> <%= community.name%> </:col>
-<:col :let={community} label="Location Area Zone"> <%= community.location_area_zone%> </:col>
-<:col :let={community} label="Edit"> 
-    <.link phx-click="delete" phx-value-id={community.id}> <span class="text-blue-500">Delete</span> </.link>    
-</:col>
+  <:col :let={community} label="Community Name"> <%= community.name%> </:col>
```
Comment (nittin-shankar) [inline lib/eco_service_web/live/eco_service_live/community_list.html.heex:3]: This should've ideally been like below:
```heex
  <:col :let={community} label="Community Name"><%= community.name %></:col>
```

Comment (nittin-shankar) [reply lib/eco_service_web/live/eco_service_live/community_list.html.heex:3]: Find the difference 😆

File: lib/eco_service_web/live/eco_service_live/community_list.html.heex [hunk 3]
```diff
@@ -1,14 +1,35 @@
...
+  <:col :let={community} label="Location Area Zone"> <%= community.location_area_zone%> </:col>
```
Comment (nittin-shankar) [inline lib/eco_service_web/live/eco_service_live/community_list.html.heex:4]: This line should've ideally been like below:
```
<:col :let={community} label="Location Area Zone">
  <%= community.location_area_zone %>
</:col>
```

File: lib/eco_service_web/live/eco_service_live/community_list.html.heex [hunk 4]
```diff
@@ -1,14 +1,35 @@
...
 <% end%>
 <%= if @all_records - @offset > @limit do %>
 <.link phx-click="next" ><p class="text-blue-500">Next</p></.link>
-<% end %>
\ No newline at end of file
+<% end %>
+
+
+<.modal
+  :if={@live_action in [:edit]}
```
Comment (nittin-shankar) [inline lib/eco_service_web/live/eco_service_live/community_list.html.heex:22]: The `in` operator is particularly useful if there are multiple items in the list. The list that you have, contains only one item. How do you think you can write this better?

File: lib/eco_service_web/live/eco_service_live/edit_community_component.ex [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      </div>
+    """
+  end
+
+  def update(assigns, socket) do
+    community = EcoServiceContext.get_community_by_id(assigns.community_id)
+
+    {:ok,
+    socket
+    |> assign(:community, community)
```
Comment (nittin-shankar) [inline lib/eco_service_web/live/eco_service_live/edit_community_component.ex:23]: Using just `|>` operator for a function is generally not considered as a good practice. There is a way that you can write this without it. Can you guess?

File: lib/eco_service_web/live/eco_service_live/edit_community_component.ex [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    }
+  end
+
+  def handle_event("update-community", params, socket) do
+    community = EcoServiceContext.update_community(socket.assigns.community, params)
+
+    case community do
```
Comment (nittin-shankar) [inline lib/eco_service_web/live/eco_service_live/edit_community_component.ex:30]: Why use `case` here at all? Is this code incomplete by any chance? I highly think it's incomplete. If you still think this code is complete, then, please have a look at the default generated code for a reference of best practice. If you're still unsure, then schedule a 10-15 minute meeting regarding this matter.

File: lib/eco_service_web/live/eco_service_live/edit_community_component.ex [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      {:ok, _} ->
+      {:noreply,
```
Comment (nittin-shankar) [inline lib/eco_service_web/live/eco_service_live/edit_community_component.ex:32]: This line should've been indented by two spaces.

**Review by nittin-shankar — COMMENTED** [review]
Let me first share my two cents about your commits:
- `Delete records` could have been `Feature to delete community and waste`. This is more descriptive.
- `Edit communities` could have been `Feature to update community details`. 

I'm already seeing some great improvement in the way you write commit messages. No grammar mistakes in your commits today. Yay! 

The next time you assign me as a reviewer for a PR, make sure you I have write access to the repository. I'm unable to use the feature called suggestions. You can watch about it [here](https://egghead.io/lessons/github-add-suggestions-in-a-github-pr-review).

Make sure you check all the comments that I've given to your code. Some of the comments I have given are not immediate answers to the problems. I've given them in a way to make you think. I have a feeling you'll be able to figure it out. Good luck 😄 

Please let me know if you find it too overwhelming and need my help. I'd be more happy to sit with you and help you think solutions.

File: lib/eco_service_web/live/eco_service_live/community_list.html.heex (hunk 4 above)
Comment (sangeethailango) [reply lib/eco_service_web/live/eco_service_live/community_list.html.heex:22]: = ?

File: lib/eco_service_web/live/eco_service_live/edit_community_component.ex (hunk 5 above)
Comment (sangeethailango) [reply lib/eco_service_web/live/eco_service_live/edit_community_component.ex:23]: assign(socket, :community, community) ?

File: lib/eco_service_web/live/eco_service_live/edit_community_component.ex (hunk 6 above)
Comment (sangeethailango) [reply lib/eco_service_web/live/eco_service_live/edit_community_component.ex:30]: I didn't match for error. Is that what you are trying to say?

**Pull request merged**